pkg/controller/repositories/repository.go

```diff
@@ -0,0 +1,203 @@
+	currentSpec := cr.Spec.ForProvider.DeepCopy()
+	repositories.LateInitialize(&cr.Spec.ForProvider, *r)
+	if !cmp.Equal(currentSpec, &cr.Spec.ForProvider) {
+		if err := e.client.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errKubeUpdateRepository)
+		}
+		lateInit = true
+	}
+
+	cr.Status.SetConditions(xpv1.Available())
+	cr.Status.AtProvider = repositories.GenerateObservation(*r)
+
+	upToDate, err := repositories.IsUpToDate(&cr.Spec.ForProvider, r)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errCheckUpToDate)
+	}
+
+	return managed.ExternalObservation{
+		ResourceUpToDate:        upToDate,
+		ResourceExists:          true,
+		ResourceLateInitialized: lateInit,
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mgd.(*v1alpha1.Repository)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	repo := &github.Repository{}
+	repositories.GenerateRepository(cr.Spec.ForProvider, repo)
+	_, _, err := e.gh.Create(
+		ctx,
+		ghclient.StringValue(cr.Spec.ForProvider.Organization),
+		repo,
+	)
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateRepository)
+	}
+
+	cr.SetConditions(xpv1.Creating())
+
+	return managed.ExternalCreation{}, nil
+}
+
+func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) { // nolint:gocyclo
+	cr, ok := mgd.(*v1alpha1.Repository)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+
+	r, _, err := e.GetRepository(
+		ctx,
+		cr.Spec.ForProvider.Owner,
+		cr.Spec.ForProvider.Name,
+		cr.Status.AtProvider.Name,
+	)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errGetRepository)
+	}
+
+	repositories.GenerateRepository(cr.Spec.ForProvider, r)
+
+	_, _, err = e.gh.Edit(
+		ctx,
+		cr.Spec.ForProvider.Owner,
+		cr.Status.AtProvider.Name,
+		r,
+	)
+	return managed.ExternalUpdate{}, errors.Wrap(err, errUpdateRepository)
+}
+
+func (e *external) Delete(ctx context.Context, mgd resource.Managed) error {
+	cr, ok := mgd.(*v1alpha1.Repository)
+	if !ok {
+		return errors.New(errUnexpectedObject)
+	}
+
+	_, err := e.gh.Delete(ctx,
+		cr.Spec.ForProvider.Owner,
+		cr.Spec.ForProvider.Name,
+	)
+	return errors.Wrap(err, errDeleteRepository)
+}
+
+// GetRepository makes API calls to get the Repository.
+// If using the Spec name the repository is not found, a second attempt
+// is made with the status name. This is useful when updating the Repository name.
+func (e *external) GetRepository(ctx context.Context, owner, specName, statusName string) (*github.Repository, *github.Response, error) {
+	repo, res, err := e.gh.Get(ctx, owner, specName)
+	if err != nil {
+		repo, res, err = e.gh.Get(ctx, owner, statusName)
+		if err != nil {
+			return nil, res, err
+		}
+		return repo, res, nil
+	}
+	return repo, res, nil
```

Comment:
Acho que fica melhor fazer um early return, evita esse encadeamento de IFs.

```suggestion
	if err == nil {
	      return repo, res, nil
	}
      repo, res, err = e.gh.Get(ctx, owner, statusName)
      if err != nil {
	      return nil, res, err
      }
      return repo, res, nil
}
```

Reply:
Fica melhor mesmo!